src/utils/compatibility.ts

```diff
@@ -4,38 +4,29 @@ import { Notification, Source as MessageTraySource } from '@girs/gnome-shell/dis
 import St from '@girs/st-16';
 
 // compatibility functions to check if a specific gnome-shell is used
-export function isGnomeVersion(version: number): boolean {
-  const [major, _minor, _patch, ..._rest]: Array<number | undefined> = PACKAGE_VERSION.split('.').map((num) => {
-    const result = parseInt(num);
-    if (isNaN(result)) {
-      return undefined;
-    }
-    return result;
-  });
-
-  if (major === undefined) {
-    return PACKAGE_VERSION.includes(version.toString());
+const GNOME_VERSION = PACKAGE_VERSION.split('.').reduce((acc, str): number[] => {
+  const result = parseInt(str);
+  if (isNaN(result)) {
+    return acc;
   }
 
-  return major === version;
-}
+  return [...acc, result];
+}, [] as number[]);
 
-export function isOneGnomeVersion(versions: number[]): boolean {
-  for (const version of versions) {
-    const isVersion = isGnomeVersion(version);
-    if (isVersion) {
-      return true;
-    }
+export function isGnomeVersionOrHigher(version: number): boolean {
+  if (GNOME_VERSION.length < 1) {
```

Comment:
Guarantees are guaranteed...

There is no reason to check what will always be guaranteed by a correct gnome-shell.

Someone would have to go out of their way to build and install gnome-shell with an "invalid" version, in which case, any issue is their fault.

Reply:
I know, but it adds little cost and I like to explicitly check for possible errors, if somewhere in the future this changes out of nowhere, for any reason whatsoever, its there to guard the rest of the code and give a better error message.

Reply:
The PACKAGE_VERSION constant has been around for 14 years. That's probably older than some of this extension's users!

It's not going anywhere, and if it is, manual intervention will be needed regardless because something like the GNOME 47+ check will start failing.